examples/text_labels_and_annotations/date.py

```diff
@@ -36,33 +33,42 @@
 # column.
 data = cbook.get_sample_data('goog.npz', np_load=True)['price_data']
 
-fig, ax = plt.subplots()
-ax.plot('date', 'adj_close', data=data)
+fig, axs = plt.subplots(3, 1, figsize=(6.4, 7), constrained_layout=True)
+# common to all three:
+for ax in axs:
+    ax.plot('date', 'adj_close', data=data)
+    # Round to nearest years.
+    datemin = np.datetime64(data['date'][0], 'Y')
+    datemax = np.datetime64(data['date'][-1], 'Y') + np.timedelta64(1, 'Y')
+    # Minor ticks every month.
+    fmt_month = mdates.MonthLocator()
+    ax.xaxis.set_minor_locator(fmt_month)
+    fmt_half_year = mdates.MonthLocator(interval=6)
+    ax.xaxis.set_major_locator(fmt_half_year)
+
+    ax.set_xlim(datemin, datemax)
+    ax.grid(True)
+    ax.set_ylabel(r'Price [\$]')
+    # Format the coords message box, i.e. the numbers displayed as the cursor
+    # moves across the axes within the interactive GUI.
+    ax.format_ydata = lambda x: f'${x:.2f}'  # Format the price.
```

Comment:
```suggestion
```
Let's throw this out. It has nothing to do with dates formatting and only clutters the example.